README.md

```diff
@@ -9,9 +9,213 @@ Ballerina SOAP Library
 
-Soap module abstracts out the details of the creation of a SOAP envelope, headers, and the body in a SOAP message.
+SOAP module abstracts out the details of the creation of a SOAP envelope, headers, and the body in a SOAP message.
+
+## Client
+
+The `Client` is used to connect to and interact with `SOAP` endpoints.
+
+### SOAP 1.1 Client
+
+```ballerina
+import ballerina/soap:soap11;
+
+soap11:Client soapClient = check new ("http://www.dneonline.com/calculator.asmx?WSDL");
+```
+
+### SOAP 1.2 Client
+
+```ballerina
+import ballerina/soap:soap12;
+
+soap12:Client soapClient = check new ("http://www.dneonline.com/calculator.asmx?WSDL");
+```
+
+## APIs associated with SOAP
+
+- **Send & Receive**: Sends SOAP request and receives a response.
+- **Send Only**: Fires and forgets requests. Sends the request without the possibility of any response from the service.
+
+The SOAP 1.1 specification requires the inclusion of the `action` parameter as a mandatory component within its APIs. In contrast, SOAP 1.2 relaxes this requirement, making the action parameter optional.
+
+### Example: Send & Receive
+
+```ballerina
+import ballerina/soap:soap11;
+
+public function main() returns error? {
+    soap11:Client soapClient = check new ("http://www.dneonline.com/calculator.asmx?WSDL");
+
+    xml envelope = xml `<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" soap:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
+                            <soap:Body>
+                            <quer:Add xmlns:quer="http://tempuri.org/">
+                                <quer:intA>2</quer:intA>
+                                <quer:intB>3</quer:intB>
+                            </quer:Add>
+                            </soap:Body>
+                        </soap:Envelope>`;
+    xml|mime:Entity[] response = check soapClient->sendReceive(envelope, "http://tempuri.org/Add");
+}
+```
+
+### Example: Send Only
+
+```ballerina
+import ballerina/soap:soap11;
+
+public function main() returns error? {
+    soap11:Client soapClient = check new ("http://www.dneonline.com/calculator.asmx?WSDL");
+
+    xml envelope = xml `<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" soap:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
+                            <soap:Body>
+                            <quer:Add xmlns:quer="http://tempuri.org/">
+                                <quer:intA>2</quer:intA>
+                                <quer:intB>3</quer:intB>
+                            </quer:Add>
+                            </soap:Body>
+                        </soap:Envelope>`;
+    xml|mime:Entity[] response = check soapClient->sendOnly(envelope, "http://tempuri.org/Add");
```

Comment:
```suggestion
    check soapClient->sendOnly(envelope, "http://tempuri.org/Add");
```

Reply:
Fixed in [1de6750](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/1de6750f3a32d73e90b0ade50476774f2541fccf)